test-data/unit/check-generics.test

```diff
@@ -1296,6 +1296,16 @@ a = A().g(1)
 a = 1
 a = '' # E: Incompatible types in assignment (expression has type "str", variable has type "int")
 
+[case testGenericInnerClass]
+from typing import TypeVar, Generic
+T = TypeVar('T')
+class A:
+    class B(Generic[T]):
+        pass
+    B[int]()
+
+A.B[int]()
+
```

Comment:
Here I would also add tests that add a generic method to inner class and then verify that type variables are substituted correctly, something like:
```python
class A:
    class B(Generic[T]):
        def meth(self) -> T:
            ...
reveal_type(A.B[int]().meth()) # This should be 'int'
```
etc.